packages/nuxt/src/runtime/hooks/wrapMiddlewareHandler.ts

```diff
@@ -0,0 +1,122 @@
+import {
+  type RequestEventData,
+  type SpanAttributes,
+  captureException,
+  debug,
+  flushIfServerless,
+  getClient,
+  getDefaultIsolationScope,
+  getIsolationScope,
+  httpHeadersToSpanAttributes,
+  httpRequestToRequestData,
+  SEMANTIC_ATTRIBUTE_SENTRY_OP,
+  SEMANTIC_ATTRIBUTE_SENTRY_ORIGIN,
+  SEMANTIC_ATTRIBUTE_SENTRY_SOURCE,
+  SPAN_STATUS_ERROR,
+  SPAN_STATUS_OK,
+  startSpan,
+  withIsolationScope,
+} from '@sentry/core';
+import type { EventHandler, EventHandlerRequest, H3Event } from 'h3';
+
+/**
+ * Wraps a middleware handler with Sentry instrumentation.
+ *
+ * @param handler The middleware handler.
+ * @param fileName The name of the middleware file.
+ */
+export function wrapMiddlewareHandler(handler: EventHandler, fileName: string) {
+  return async (event: H3Event<EventHandlerRequest>) => {
+    debug.log(`Sentry middleware: ${fileName} handling ${event.path}`);
+
+    const isolationScope = getIsolationScope();
+    const newIsolationScope = isolationScope === getDefaultIsolationScope() ? isolationScope.clone() : isolationScope;
```

Comment:
let's verify if we actually need this :) we have this code in a bunch of places (which is, I assume, where you got this from xD) but possibly this is not needed here. Is there ever a case where this is called without it being wrapped by a http.server call that already isolates the span? 🤔